python_modules/libraries/dagster-aws/dagster_aws/ecs/executor.py

```diff
@@ -0,0 +1,333 @@
+import json
+import os
+from typing import Any, List, Mapping, Optional, Sequence, cast
+
+import boto3
+from dagster import (
+    DagsterRun,
+    Field,
+    IntSource,
+    Permissive,
+    _check as check,
+    executor,
+)
+from dagster._core.definitions.executor_definition import multiple_process_executor_requirements
+from dagster._core.definitions.metadata import MetadataValue
+from dagster._core.events import DagsterEvent, EngineEventData
+from dagster._core.execution.retries import RetryMode, get_retries_config
+from dagster._core.execution.tags import get_tag_concurrency_limits_config
+from dagster._core.executor.base import Executor
+from dagster._core.executor.init import InitExecutorContext
+from dagster._core.executor.step_delegating import (
+    CheckStepHealthResult,
+    StepDelegatingExecutor,
+    StepHandler,
+    StepHandlerContext,
+)
+from dagster._utils.backoff import backoff
+
+from dagster_aws.ecs.container_context import EcsContainerContext
+from dagster_aws.ecs.launcher import STOPPED_STATUSES, EcsRunLauncher
+from dagster_aws.ecs.tasks import (
+    get_current_ecs_task,
+    get_current_ecs_task_metadata,
+    get_task_kwargs_from_current_task,
+)
+from dagster_aws.ecs.utils import RetryableEcsException, run_ecs_task
+
+DEFAULT_STEP_TASK_RETRIES = "5"
+
+
+_ECS_EXECUTOR_CONFIG_SCHEMA = {
+    "run_task_kwargs": Field(
+        Permissive({}),
+        is_required=False,
+        description=(
+            "Additional arguments to include while running the task. See"
+            " https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/ecs.html#ECS.Client.run_task"
+            " for the available parameters. The overrides and taskDefinition arguments will"
+            " always be set by the run launcher."
+        ),
+    ),
+    "cpu": Field(IntSource, is_required=False),
+    "memory": Field(IntSource, is_required=False),
+    "ephemeral_storage": Field(IntSource, is_required=False),
+    "task_overrides": Field(
+        Permissive({}),
+        is_required=False,
+    ),
```

Comment:
recognizing that this was all on my original PR - there's some overlap between these parameters right? All of them can be set on run_task_kwargs. I think it makes sense to put things like cpu/memory as top-level fields since they are fields that most people will want to change

What happens if you set 'overrides' on the run_task_kwargs param though? should we not allow that? should we merge them? how do we decide which one wins if you set both "task_overrides.cpu" and "cpu"? What if you set task_overrides.containerOverrides?

the snake case vs. camel case thing is kind of a bummer too (dagster is generally snake case but the boto apis expect camelCase, so you end up with grossness like task_overrides.containerOverrides), but I don't see an obvious way around that